src/tests/Feature/EventTest.php

```diff
@@ -4,15 +4,35 @@
 
 use App\Models\Event;
 
+
 class EventTest extends PermissionsTest
 {
+    use ResourceWithCoordinatesTrait;
     /**
      * The parent class will test that the user can access the page and that the user can't access the page if he doesn't have the permission.
      */
     // set up the ids array
     public function setUp(): void {
         parent::setUp();
-        $this->route = "event";
+        $this->route = 'event';
         $this->resource = Event::factory()->create();
     }
+
+    /**
+     * @param $new_item
+     * @return mixed
+     */
+    protected function customizeStoreTest($new_item){
+        $new_item["event_subtype_id"] = null;
+        return $new_item;
+    }
+
+    /**
+     * @param $new_item
+     * @return mixed
+     */
+    protected function customizeUpdateTest($new_item) {
+        $new_item["event_subtype_id"] = $this->resource->event_subtype_id; // subtype is not available yet
```

Comment:
Can you provide a use case for those functions ? (please)

Reply:
That's kinda tricky.
I want to test storing and updating data using the controller routes.
Event and Places are using coordinate. The data should have a specific format. (check the Trait).

The trait allow to handle this specific format. The purpose of the trait is to have a global code for each model with coordinates element. However, sometimes, your model needs specific data.

This is the case of Event. Event needs to add 'event_subtype_id' so I overwrite cuztomizeUpdateTest function defined in the Trait.